client/src/app/pages/applications/components/application-detail-drawer/components/download-button.tsx

```diff
@@ -1,32 +1,19 @@
-import React, { useState } from "react";
-import { Alert, Button } from "@patternfly/react-core";
+import React from "react";
+import { Alert, Button, Spinner } from "@patternfly/react-core";
 import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
-import { Application } from "@app/api/models";
-import { Spinner } from "@patternfly/react-core";
-import { useDownloadStaticReport } from "@app/queries/download";
+import { saveAs } from "file-saver";
 
-export enum MimeType {
-  TAR = "tar",
-  YAML = "yaml",
-}
-function DownloadButton({
-  application,
-  mimeType,
-}: {
-  application: Application;
-  mimeType: MimeType;
-}) {
-  const {
-    mutate: downloadFile,
-    isLoading,
-    isError,
-  } = useDownloadStaticReport();
+import { MimeType } from "@app/api/models";
+import { useFetchStaticReport } from "@app/queries/applications";
+
+function DownloadButton({ id, mimeType }: { id: number; mimeType: MimeType }) {
```

Comment:
Could extract this to a defined interface or type:

```suggestion
function DownloadButton({ id, mimeType }: DownloadButtonProps) {
```

Reply:
Why `function` syntax here and not a const using the `React.FC` type?

Reply:
Ah I realized that's preexisting. Maybe we should take this opportunity to make it consistent with our other components